### ISSUE TITLE
Python prompt_toolkit v2

### ISSUE THREAD

**Issue by zhutony** [facebookincubator/python-nubia#22]
has any work been done to upgrade nubia to use prompt_toolkit v2?

**Comment by ylectric** [comment]
@zhutony not yet, but we're considering this. It's not a top priority, although it's on our mid-term roadmap. Any help would be appreciated.

**Comment by zhutony** [comment]
Thanks a lot for your quick response. I already made it work in an application specific way. Although I have no idea how much more work needed to generalize my code to be in Nubian way, I'm interested in doing it. Any suggestions or related documents are appreciated.

**Comment by AhmedSoliman** [comment]
@zhutony Can you explain a bit more what exactly are you looking for?

**Comment by zhutony** [comment]
that's actually a good question I had asked to myself:-) and the short answer is there is no real reason.

longer version:
after having played a while with 'original' Nubia, I really liked it and convinced my team to use it. I was worried about the cases when some widgets from Python Prompt Toolkit are needed in future and I started to dig into the toolkit a bit. The toolkit v1 seems to be old and jumping on v2 would make me worry less for a new project.

So I upgraded/hacked the toolkit from v1 to v2, but in an application specific manner - just wanted to have a quick prototype of Nubia on v2. But the upgrade is not my priority any more because recently we've decided to stay on v1 for a while first.
